ament_cpplint/ament_cpplint/main.py

```diff
@@ -144,7 +148,9 @@ def main(argv=sys.argv[1:]):
         else:
             print("Not using '--root'")
         print('')
-        arguments += files
+        arguments += [
+            f for f in files
+            if any(excl not in f or not excl for excl in args.exclude)]
```

Comment:
This logic is still wrong. You want to check `if f not in args.excludes`.

Reply:
`args.exclude` constains substring of `f`. This condicion is not valid `if f not in args.excludes`.

`args.exclude` could contain directories which make the condition is not valid.

``
f = /tmp/ros2_foxy/src/ros2/geometry2/tf2/test/cache_unittest.cpp
args.excludes = test
```

In this case using this condition f is not in args.excludes. This is True when we want False (to not include this file)

but `any(excl not in f for excl in args.exclude)`  the substring is compared with the full name which will return True when the substring it not in the filename (adding this file to the cpplint argument list) and False when it is (removing it from the argument list)